JS/Airport.js

```diff
@@ -1,120 +1,64 @@
     }
 
-
-
-
-
-
-
-    getTransportMilitaryPlanes(){
-        let transportMilitaryPlanes = [];
-        let militaryPlanes = this.getMilitaryPlanes();
-        for (let i = 0; i < militaryPlanes.length; i++) {
-        if (militaryPlanes[i].getMilitaryType() == MilitaryType.TYPE_TRANSPORT) {
-        transportMilitaryPlanes.push(militaryPlanes[i]);
-        }
-        }
-        return transportMilitaryPlanes;
+    getMilitaryPlanes() {
+        const militaryPlanes = this.planes.filter(plane => {
+            return plane instanceof MilitaryPlane;
+        })
+        return militaryPlanes;
     }
 
-
-    getBomberMilitaryPlanes()
-    {
-        let bomberMilitaryPlanes = [];
-        let militaryPlanes = this.getMilitaryPlanes();
-        for (let i = 0; i < militaryPlanes.length; i++) {
-            if (militaryPlanes[i].getMilitaryType()=== MilitaryType.BOMBER) {
-                bomberMilitaryPlanes.push(militaryPlanes[i]);
-            }
-        }
-        return bomberMilitaryPlanes;
+    getTransportMilitaryPlanes() {
+        const getTransportMilitaryPlanes = this.planes.filter(plane => {
+            return plane.militaryType === militaryType.transport;
+        })
+        // console.log(getTransportMilitaryPlanes)
+        return getTransportMilitaryPlanes;
     }
 
-    constructor(planes) {
-        this.planes = planes;
+    getBomberMilitaryPlanes() {
+        const getBomberMilitaryPlanes = this.planes.filter(plane => {
+            return plane.militaryType === militaryType.bomber;
+        })
+        return getBomberMilitaryPlanes;
     }
 
-
     getExperimentalPlanes() {
-        let experimentalPlanes  = [];
-        this.planes.forEach(plane => {
-            if (plane instanceof experimentalPlane) {//if
-                experimentalPlanes.push(plane);
-            }
-        });
-//return
-        return experimentalPlanes;
-    }
-
-
-
-    sortByMaxDistance() {
-        this.planes.sort((a, b) => (a.Get_Max_Flight_Distance() > b.Get_Max_Flight_Distance()) ? 1 : -1);
-        return this;
-    }
-
-    /**
-     * Sorts by max speed
-     * @return Airport
-     */
-    sortByMaxSpeed() {
-        this.planes.sort((a, b) => (a.getMS() > b.getMS()) ? 1 : -1);
-        return this;
+        const getExperimentalPlanes = this.planes.filter(plane => {
+            return plane instanceof experimentalPlane;
+        })
+        return getExperimentalPlanes;
     }
 
     sortByMaxLoadCapacity() {
-        this.planes.sort((a, b) => (a.getMinLoadCapacity() > b.getMinLoadCapacity()) ? 1 : -1);
-        return this;
-    }
-
-    getPlanes() {
-        return this.planes;
-    }
-
-
-
-
-    static print(planes) {
-        return JSON.stringify(planes);
+        const sortByMaxLoadCapacity = this.planes.sort((a, b) => (a.maxLoadCapacity - b.maxLoadCapacity));
+        return sortByMaxLoadCapacity;
     }
 }
-
-module.exports = Airport;
+module.exports = Airport;
```

Comment:
The best practice is to use the following syntax while exporting variables from the file:
`module.exports = { Airport }`

Reply:
Returns: TypeError: Airport is not a constructor